src/types/string.rs

```diff
@@ -298,6 +298,13 @@ impl<'a> IntoPy<PyObject> for &'a str {
     }
 }
 
+impl<'a> IntoPy<Py<PyString>> for &'a str {
+    #[inline]
+    fn into_py(self, py: Python<'_>) -> Py<PyString> {
+        PyString::new(py, self).into()
+    }
+}
+
```

Comment:
This is a (mild) breaking change because of inference errors, but not too bad.

Reply:
Just to make sure I understand things correctly: The comment is informational as the other changes are breaking anyway and this is therefore targeting 0.17.0?

Reply:
Is the previous instance into PyObject even useful anymore?

Reply:
> Is the previous instance into PyObject even useful anymore?

I think so as `Py<PyString>` does not automatically "decay" into `Py<PyAny>` and the existing impl might be used in generic context to produce `PyObject` for stashing away.

Reply:
Easy to get PyObject from a Py though.

Reply:
> Easy to get PyObject from a Py though.

I think that in generic context, one needs to be able to express this in a single trait bound, e.g. calling `into_py` twice or `into_py` followed by `into` to go from `T` to `Py<S>` to `PyObject` is not necessarily the same as going from `T` to `PyObject`.

Reply:
> Just to make sure I understand things correctly: The comment is informational as the other changes are breaking anyway and this is therefore targeting 0.17.0?

Mainly it means that things like `let thing = "thing".into_py(py);` stop compiling

Reply:
To avoid making lots of annoying inference errors we could potentially combine this PR with #2316, which would enable us to select on types which have `IntoPyObject<Target = PyString>` instead of `IntoPy<Py<PyString>>`.